include/oneapi/dpl/pstl/utils.h

```diff
@@ -299,37 +314,41 @@ class __set_value
 
 //TODO: to do the same fix  for output type (by re-using __transform_functor if applicable) for the other functor below:
 // __transform_if_unary_functor, __transform_if_binary_functor, __replace_functor, __replace_copy_functor
-//TODO: to make input type consistently: const T& or T&&; to think which way is preferable
-template <typename _Pred>
+template <typename _F, typename _RevTag = std::false_type>
 class __transform_functor
 {
-    mutable _Pred _M_pred;
+    mutable _F __f;
 
   public:
-    explicit __transform_functor(_Pred __pred) : _M_pred(::std::move(__pred)) {}
+    explicit __transform_functor(_F __f) : __f(std::move(__f)) {}
 
     template <typename _Input1Type, typename _Input2Type, typename _OutputType>
     void
-    operator()(const _Input1Type& __x, const _Input2Type& __y, _OutputType&& __output) const
+    operator()(_Input1Type&& __x, _Input2Type&& __y, _OutputType&& __output) const
```

Comment:
I have a memory that using const lvalue ref here for inputs instead of universal references was intentional but I don't remember why.  I think #1233 was the last time we considered this in detail. I'm looking through that PR now to try to remind myself.